drivers/video/video_esp32_dvp.c

```diff
@@ -359,6 +423,13 @@ static int video_esp32_init(const struct device *dev)
 		return -ENODEV;
 	}
 
+#ifdef CONFIG_VIDEO_ESP32_VSYNC_INTERRUPT
+	if (!video_esp32_enable_interrupt(dev)) {
+		LOG_ERR("Failed to enable interrupt");
+		return -EIO;
+	}
```

Comment:
Is that not swapped? 0 means success in ESP-IDF like in Zephyr?
https://github.com/espressif/esp-idf/blob/81f4ef73c358e924e1e34bed0e51b5989dcfefa3/components/esp_common/include/esp_err.h#L22

You can also convert it to the following idiom to help proofreading the error codes.

```suggestion
int ret;
...
ret = video_esp32_enable_interrupt(dev);
if (ret) {
	LOG_ERR("Failed to enable interrupt");
	return ret;
}
```